### PR TITLE
App Creation

### PR DESCRIPTION
**Pull request by shrinathprabhu**
Resolves [AR-2448](https://team-1624093970686.atlassian.net/browse/AR-2448)

## Changes

- Added create app component and integrated it with api
- Added empty app name validation while create
- Updated delete app api and moved it to gateway service from app config service
- Renamed routes from `/configure` to `/app/configure` so we can also have `/app/create` and in future we can have `/app/:appId/configure`

## Screenshots
Screenshot from 2022-05-17 06-52-30
Screenshot from 2022-05-17 06-52-37

## Checklist

- [x] The branch name follows the format: `developer/AR-XXX-issue-name`.
- [x] The changes have been tested locally.

### CODE REVIEW

File: src/components/app-configure/CreateApp.vue [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import VStack from '@/components/lib/VStack/VStack.vue'
+import VTextField from '@/components/lib/VTextField/VTextField.vue'
+import { createApp } from '@/services/gateway.service'
+
+const store = useStore()
+const router = useRouter()
+const appName: ComputedRef<string> = computed(() => store.getters.appName)
+const appNameError: Ref<boolean> = ref(false)
+
+function onAppNameChange(appName: string) {
```
Comment (ajithranka) [inline src/components/app-configure/CreateApp.vue:19]:
```suggestion
function handleAppNameChange(appName: string) {
```

File: src/components/app-configure/CreateApp.vue [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+import VOverlay from '@/components/lib/VOverlay/VOverlay.vue'
+import VSeperator from '@/components/lib/VSeperator/VSeperator.vue'
+import VStack from '@/components/lib/VStack/VStack.vue'
+import VTextField from '@/components/lib/VTextField/VTextField.vue'
+import { createApp } from '@/services/gateway.service'
+
+const store = useStore()
+const router = useRouter()
+const appName: ComputedRef<string> = computed(() => store.getters.appName)
+const appNameError: Ref<boolean> = ref(false)
```
Comment (ajithranka) [inline src/components/app-configure/CreateApp.vue:17]:
```suggestion
const hasAppNameError: Ref<boolean> = ref(false)
```

File: src/components/app-configure/CreateApp.vue [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        label="CREATE"
+        class="create-button"
+        @click.stop="handleCreateApp"
+      />
+    </VCard>
+  </section>
+</template>
+
+<style scoped>
+section {
```
Comment (ajithranka) [inline src/components/app-configure/CreateApp.vue:61]: Prefer using class names in component styles over element selectors.

File: src/components/app-configure/CreateApp.vue [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+          :message-type="appNameError ? 'error' : ''"
+          message="App Name cannot be empty"
+          @update:model-value="onAppNameChange"
+          @keyup.enter="handleCreateApp"
+        />
+      </VStack>
+      <VButton
+        label="CREATE"
+        class="create-button"
+        @click.stop="handleCreateApp"
```
Comment (ajithranka) [inline src/components/app-configure/CreateApp.vue:54]: Disable when app name is empty.

**Review by karthik-durai — COMMENTED** [review]
please make request changes.

**Review by ajithranka — APPROVED** [review]

**Review by karthik-durai — APPROVED** [review]

**Pull request merged**